engine/src/cmd/basecomputer.cpp

```diff
@@ -4113,43 +4103,36 @@ void trackPrice(int whichplayer, const Cargo &item, float price, const string &s
 
         BOOST_LOG_TRIVIAL(info) << "Tracking data:";
         BOOST_LOG_TRIVIAL(info) << boost::format("  highest locs: (%1%)") % recordedHighestLocs.size();
-        // VSFileSystem::vs_dprintf(1,"Tracking data:\n");
-        // VSFileSystem::vs_dprintf(1,"  highest locs: (%d)\n", recordedHighestLocs.size());
         {
             for (size_t i = 0; i < recordedHighestLocs.size(); ++i) {
                 BOOST_LOG_TRIVIAL(info) << boost::format("    %1% : %2%") % i % recordedHighestLocs[i];
-                // VSFileSystem::vs_dprintf(1, "    %d : %s\n", i, recordedHighestLocs[i].c_str());
             }
         }
 
-        //BOOST_LOG_TRIVIAL(info) << boost::format("  highest prices: (%1%)") % recordedHighestPrices.size();
-        VSFileSystem::vs_dprintf(1,"  highest prices: (%d)\n", recordedHighestPrices.size());
+        BOOST_LOG_TRIVIAL(info) << boost::format("  highest prices: (%1%)") % recordedHighestPrices.size();
         {
             for (size_t i = 0; i < recordedHighestPrices.size(); ++i) {
+                                                                    // POSIX-printf style
                 BOOST_LOG_TRIVIAL(info) << boost::format("    %1% : %2$.2f") % i % recordedHighestPrices[i];
-                // VSFileSystem::vs_dprintf(1, "    %d : %.2f\n", i, recordedHighestPrices[i]);
             }
         }
 
-        //BOOST_LOG_TRIVIAL(info) << boost::format("  lowest locs: (%1%)") % recordedLowestLocs.size();
-        VSFileSystem::vs_dprintf(1,"  lowest locs: (%d)\n", recordedLowestLocs.size());
+        BOOST_LOG_TRIVIAL(info) << boost::format("  lowest locs: (%1%)") % recordedLowestLocs.size();
         {
             for (size_t i = 0; i < recordedLowestLocs.size(); ++i) {
                 BOOST_LOG_TRIVIAL(info) << boost::format("    %1% : %2%") % i % recordedLowestLocs[i];
-                // VSFileSystem::vs_dprintf(1, "    %d : %s\n", i, recordedLowestLocs[i].c_str());
             }
         }
 
-        //BOOST_LOG_TRIVIAL(info) << boost::format("  lowest prices: (%1%)") % recordedLowestPrices.size();
-        VSFileSystem::vs_dprintf(1,"  lowest prices: (%d)\n", recordedLowestPrices.size());
+        BOOST_LOG_TRIVIAL(info) << boost::format("  lowest prices: (%1%)") % recordedLowestPrices.size();
         {
             for (size_t i = 0; i < recordedLowestPrices.size(); ++i) {
+                                                                    // POSIX-printf style
                 BOOST_LOG_TRIVIAL(info) << boost::format("    %1% : %2$.2f") % i % recordedLowestPrices[i];
-                // VSFileSystem::vs_dprintf(1, "    %d : %.2f\n", i, recordedLowestPrices[i]);
             }
         }
 
-        fflush(stderr);
+        // TODO: Find a way to flush the Boost logs? Console log specifically?
```

Comment:
part of the boost log configuration IIRC

Reply:
@BenjamenMeyer yes, you can set the log sinks to auto-flush after every line of output. I have done that with the console logger, but my thinking is that that will be too much overhead with the file logger; it will slow down the game too much. So we will need a way to manually flush the file log sink when needed.

Luckily, I now have a way to do that:

```C++
VSFileSystem::pFileLogSink->flush();
```

I just haven't updated this file yet to use it.

Reply:
Thanks for the review, by the way.